backend/controllers/mainController.js

```diff
@@ -0,0 +1,140 @@
+/*
+ * ====================================================================================================================
+ * This controller handles our routing functions. 
+ * We can use response.send() to send a message back to the client
+ * We can use request.params to see url paramaters (paramaters that are in the url, e.g. the "1" in api/entries/1)
+ * We can use request.body to see POST paramaters (paramaters that are sent in the POST request)
+ * ====================================================================================================================
+ */
+
+
+// First we load our dependencies. 
+var mongoose = require('mongoose'); //We need the mongoose library so we can work with our database.
+var Movie = require('../models/movie'); //We also need to load the Movie model.
+
+
+/*
+ * The table below lists the routes we will implement, just for reference.
+ * 
+ * Application Routes
+ * ------------------------------------------------------------------------------------------
+ * URI 						| HTTP ACTION 		| POST Body 		| Result
+ * ------------------------------------------------------------------------------------------
+ * api/movies 	 			GET 			 	 Empty 				  List of all users
+ * api/movies/:id 			GET    			 	 Empty   			  One user w/ details
+ * api/movies 				POST 				 JSON                 Adds provided user
+ * api/movies/:id/reviews   POST  				 JSON    			  Adds provided review
+ * ------------------------------------------------------------------------------------------
+ */
+
+
+/**
+ * This setupRoutes function will create
+ * all of the routes for our server to use
+ */
+module.exports.setupRoutes = function(app) {
+	
+	/*
+	 * This function gets called when a client makes a GET request
+	 * to the root of the app, like "www.myWebsite.com/".
+	 * It just sends back a hello world message for testing.
+	 */
+	app.get('/',function(request,response) {
+		response.send("Your server is alive, and you're in the right place!");
+	});
+
+	/*
+	 * This function gets called when a client makes a GET request to "/api/movies". 
+	 * Here we send our entire data array to the client.
+	 */
+	app.get('/api/movies',function(request,response) {
```

Comment:
One really good paradigm that more people should use is versioning of APIs.

`/api/v1/movies`

This way when functionality of a production API changes applications using the older API won't break.